swift/ql/lib/codeql/swift/security/SensitiveExprs.qll

```diff
@@ -91,11 +91,15 @@ private class SensitiveVarDecl extends VarDecl {
  */
 private class SensitiveFunction extends Function {
   SensitiveDataType sensitiveType;
+  string name; // name of the function, not including the argument list.
 
-  SensitiveFunction() { this.getName().toLowerCase().regexpMatch(sensitiveType.getRegexp()) }
+  SensitiveFunction() {
+    name = this.getName().splitAt("(", 0) and
```

Comment:
I don't seem to be able to accept your suggestion ... but I agree this should be a predicate on `Function` anyway...

Reply:
Fixed.  I've adjusted the regexp, removing the unnecessary escape on the first `(`, then modifying it so it copes with edge cases a bit better (e.g. strings that don't contain a `(`) in case they are possible.  And I've opened https://github.com/github/codeql/pull/13333 for a very similar case in CPP.

Reply:
> Fixed. I've adjusted the regexp, removing the unnecessary escape on the first `(`,

Thanks! I also see that you removed the anchor from the beginning of the regex (i.e., the `^` char). This was indeed unnecessary since the semantics of `regexpCapture` is that the regex should match _the entire_ receiver. 👍

> then modifying it so it copes with edge cases a bit better (e.g. strings that don't contain a `(`) in case they are possible. And I've opened #13333 for a very similar case in CPP.

I don't think it's possible to have a function name without a parenthesis. See the `constructName` function [here](https://github.com/github/codeql/blob/main/swift/extractor/translators/DeclTranslator.cpp#L12) which we use to create the function name in the extractor.

I _think_ requiring the `(` makes the regex slightly more efficient, but I'm not enough of an expert to champion this in any way.

Reply:
The thing I don't like is that "possessive matching" is a somewhat obscure feature, and thus a little harder to understand (that's why I called it out in a comment).  But I do believe coping with strings without `(` is valuable, even if we're _sure_ we don't generate any at present - that could change in future.